File: app/src/main/java/com/popalay/cardme/ui/base/widgets/SettingTextView.java
package com.popalay.cardme.ui.base.widgets;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.AppCompatTextView;
import android.text.TextPaint;
import android.util.AttributeSet;

import com.popalay.cardme.R;

public class SettingTextView extends AppCompatTextView {

    private String settingText;
    private TextPaint paint;

    public SettingTextView(Context context) {
        super(context);
        init(context, null, 0);
    }

    public SettingTextView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs, 0);
    }

    public SettingTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs, defStyleAttr);
    }

    public String getSettingText() {
        return settingText;
    }

    public void setSettingText(String settingText) {
        this.settingText = settingText;
        invalidate();
    }

    @Override protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        canvas.save();
        canvas.translate(getPaddingLeft(), getPaddingTop());
        final float width = canvas.getWidth() - paint.measureText(settingText)
                - getPaddingEnd() - getPaddingStart();
        final float height = -paint.ascent() + paint.descent();
        canvas.drawText(settingText, width, height, paint);
        canvas.restore();
    }

    private void init(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        if (attrs != null) {
            final TypedArray a = context.getTheme().obtainStyledAttributes(
                    attrs, R.styleable.SettingTextView, defStyleAttr, 0);
            try {
                settingText = a.getString(R.styleable.SettingTextView_settingText);
            } finally {
                a.recycle();
            }
        }
        paint = new TextPaint(Paint.ANTI_ALIAS_FLAG);
        paint.setColor(ContextCompat.getColor(context, R.color.accent));
        paint.setTextSize(getTextSize());
    }
}

File: app/src/main/kotlin/com/popalay/cardme/presentation/widget/SettingTextView.kt
package com.popalay.cardme.presentation.widget

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.support.v4.content.ContextCompat
import android.support.v7.widget.AppCompatTextView
import android.text.TextPaint
import android.util.AttributeSet
import com.popalay.cardme.R

class SettingTextView(
        context: Context,
        attrs: AttributeSet? = null,
        defStyleAttr: Int = 0
) : AppCompatTextView(context, attrs, defStyleAttr) {

    constructor(context: Context, attrs: AttributeSet) : this(context, attrs, 0)

    private var settingText: String? = null
    private val textPaint: TextPaint by lazy {
        val lazyPaint = TextPaint(Paint.ANTI_ALIAS_FLAG)
        lazyPaint.color = ContextCompat.getColor(context, R.color.accent)
        lazyPaint.textSize = textSize
        lazyPaint
    }

    init {
        attrs?.let {
            val a = context.theme.obtainStyledAttributes(attrs, R.styleable.SettingTextView, defStyleAttr, 0)
            try {
                settingText = a.getString(R.styleable.SettingTextView_settingText)
            } finally {
                a.recycle()
            }
        }
    }

    fun setSettingText(settingText: String?) {
        this.settingText = settingText
        invalidate()
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        settingText?.let {
            canvas.save()
            canvas.translate(paddingLeft.toFloat(), paddingTop.toFloat())
            val width = canvas.width
                    .minus(textPaint.measureText(settingText))
                    .minus(paddingEnd)
                    .minus(paddingStart)
            val height = -textPaint.ascent() + textPaint.descent()
            canvas.drawText(settingText, width, height, textPaint)
            canvas.restore()
        }
    }
}
